queenbee/repository/package.py

```diff
@@ -1,19 +1,166 @@
+
+    digest: str
+
+    @classmethod
+    def from_resource(cls, resource: Union[Operator, Recipe], package_path: str = None, created: datetime = None):
+
+        if package_path is None:
+            if isinstance(resource, Operator):
+                base_url = 'operators'
+            elif isinstance(resource, Recipe):
+                base_url = 'recipes'
+            else:
+                raise ValueError(f'resource should be an instance of Operator or Recipe, not: {type(resource)}')
+
+            full_name = f'{resource.metadata.name}-{resource.metadata.version}.tgz'
+            package_path = os.path.join(base_url, full_name)
+
+        if created is None:
+            created = datetime.utcnow()
+
+        input_dict = resource.metadata.to_dict()
+        input_dict['digest'] = resource.__hash__
+        input_dict['created'] = created
+        input_dict['url'] = package_path
+
+        return cls.parse_obj(input_dict)
+
+
+    @classmethod
+    def _package_resource(cls, resource, repo_folder: str, path_to_readme: str = None, overwrite: bool = False):
+        
+        resource_version = cls.from_resource(resource)
+
+        resource.to_json('resource.json')
+        resource_version.to_json('version.json')
+
+        folder_path = os.path.abspath(repo_folder)
+
+        tar_path = os.path.join(
+            folder_path,
+            resource_version.url
+        )
+
+        if not overwrite:
+            if os.path.isfile(tar_path):
+                os.remove('resource.json')
+                raise ValueError(f'Cannot overwrite file at path: {tar_path}')
+
+        tar = TarFile.open(tar_path, 'w:gz')
+        tar.add('resource.json', arcname='resource.json', filter=reset_tar)
+        tar.add('version.json', arcname='version.json', filter=reset_tar)
+
+
+        if path_to_readme is not None:
+            tar.add(os.path.abspath(path_to_readme), arcname='README.md', filter=reset_tar)
+        
+        tar.close()
+
+        # Delete original resource file
+        os.remove('resource.json')
+        os.remove('version.json')
+
+        if os.path.isfile('README.md'):
+            os.remove('README.md')
+
+        return resource_version
+
+    @classmethod
+    def from_package(cls, package_path: str):
+        file_path = os.path.abspath(package_path)
+
+        tar_file = TarFile.open(file_path)
+
+        member = tar_file.getmember('version.json')
+
+        version_bytes = tar_file.extractfile(member).read()   
+
+        return cls.parse_raw(version_bytes)
+
+    @staticmethod
+    def path_to_readme(folder_path: str):
+        path_to_readme = None
+
+        readme_pattern = r'^readme\.md$'
+        
+        for file in os.listdir(folder_path):
+            res = re.match(readme_pattern, file, re.IGNORECASE)
+            if res is not None:
+                path_to_readme = os.path.join(folder_path, file)
+
+        return path_to_readme
+
+
+class OperatorVersion(ResourceVersion, OperatorMetadata):
```

Comment:
all the classes in this module are missing `docstring` 😀

Reply:
Yep! Docstrings are coming after tests. Just want to make sure everything's pretty fixed before spending time documenting everything